concent_api/core/payments/sci_callback.py

```diff
@@ -0,0 +1,260 @@
+from contextlib import closing
+import socket
+
+from bitcoin import ecdsa_raw_verify
+from ethereum.utils import sha3
+from ethereum.transactions import UnsignedTransaction
+from ethereum.transactions import Transaction
+from golem_messages.exceptions import InvalidSignature
+from golem_messages.exceptions import MessageError
+from rlp import encode
+
+from django.conf import settings
+
+from common.helpers import generate_ethereum_address_from_ethereum_public_key_bytes
+from core.constants import SCI_CALLBACK_MAXIMUM_TIMEOUT
+from core.exceptions import SCICallbackFrameError
+from core.exceptions import SCICallbackPayloadError
+from core.exceptions import SCICallbackPayloadSignatureError
+from core.exceptions import SCICallbackRequestIdError
+from core.exceptions import SCICallbackTimeoutError
+from core.exceptions import SCICallbackTransactionSignatureError
+
+from middleman_protocol.concent_golem_messages.message import SignedTransaction
+from middleman_protocol.concent_golem_messages.message import TransactionRejected
+from middleman_protocol.concent_golem_messages.message import TransactionSigningRequest
+from middleman_protocol.exceptions import MiddlemanProtocolError
+from middleman_protocol.message import ErrorFrame
+from middleman_protocol.message import GolemMessageFrame
+from middleman_protocol.stream import send_over_stream
+from middleman_protocol.stream import unescape_stream
+
+
+class RequestIDGenerator:
+
+    _request_id = 0
+
+    @classmethod
+    def generate_request_id(cls) -> int:
+        cls._request_id += 1
+        return cls._request_id
+
+
+def sci_callback(transaction: Transaction) -> Transaction:
```

Comment:
For me comments in this functions are a bit redundant:
1) functions used within are well-named and pretty self-describing
2) they have docstring which explain their behavior.

Reply:
I will remove the `flow` part as it may get outdated easily.